app/assess/data.py

```diff
@@ -0,0 +1,107 @@
+import json
+import os
+from typing import List
+
+import requests
+from app.assess.models.application import Application
+from app.assess.models.fund import Fund
+from app.assess.models.round import Round
+from app.config import APPLICATION_STORE_API_HOST
+from app.config import FLASK_ROOT
+from app.config import FUND_STORE_API_HOST
+
+
+# Fund Store Endpoints
+FUNDS_ENDPOINT = "/funds/"
+FUND_ENDPOINT = "/funds/{fund_id}"
+ROUND_ENDPOINT = "/funds/{fund_id}/round/{round_number}"
+
+# Application Store Endpoints
+APPLICATIONS_ENDPOINT = "/fund/{fund_id}?datetime_start={datetime_start}&datetime_end={datetime_end}"
+APPLICATION_ENDPOINT = "/fund/{fund_id}?application_id={application_id}"
+
+
+def get_data(endpoint: str):
+    if endpoint[:8] == "https://":
+        response = requests.get(endpoint)
+        if response.status_code == 200:
+            data = response.json()
+        else:
+            return None
+    else:
+        data = get_local_data(endpoint)
+    return data
+
+
+def get_local_data(endpoint: str):
+    api_data_json = os.path.join(
+        FLASK_ROOT, "tests", "api_data", "endpoint_data.json"
+    )
+    fp = open(api_data_json)
+    api_data = json.load(fp)
+    fp.close()
+    if endpoint in api_data:
+        return api_data.get(endpoint)
+
+
+def get_funds() -> List[Fund] | None:
+    endpoint = FUND_STORE_API_HOST + FUNDS_ENDPOINT
+    response = get_data(endpoint)
+    if response and len(response) > 0:
+        funds = []
+        for fund in response:
+            funds.append(Fund.from_json(fund))
+        return funds
+    return None
+
+
+def get_fund(fund_id: str) -> Fund | None:
+    endpoint = FUND_STORE_API_HOST + FUND_ENDPOINT.format(fund_id=fund_id)
+    response = get_data(endpoint)
+    if response and "name" in response:
```

Comment:
At risk of being exiled by the python community (sorry folx) I actually think this would be a good use case for the match case syntax introduced in python 3.10 (which we are using!)

Here is a tutorial (a straight up pep) https://www.python.org/dev/peps/pep-0636/

Reply:
Great recommendation would love to use these - thanks for the intro. Though have a question. Here we want to create a fund (regardless of whether there are any rounds), and just add rounds if there are any present, so are nested if statements actually ok here? i.e. the fund has to exist to create a round so if we matched  cases on has_rounds=true/false, we'd need two cases and to create fund in both(duplicating that statement)? Or is there a way to use (or pass) an object created in another case?